src/plugins/intel_gpu/src/plugin/ops/rnn.cpp

```diff
@@ -224,127 +228,163 @@ static void CreateLSTMCellOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v4
+
+            p.add_primitive(*op, cldnn::concatenation(concatID, { cldnn::input_info(inputCrop_id), cldnn::input_info(hiddenStr) }, 3));
+
+            cldnn::tensor fc_input_resh_tensor = { crop_tensor.batch[0], crop_tensor.spatial[0] + inStateShape.spatial[0],
+                                                crop_tensor.feature[0], crop_tensor.spatial[1]};
+            p.add_primitive(*op, cldnn::reshape(fc_input_resh_id, cldnn::input_info(concatID), fc_input_resh_tensor));
+
+            p.add_primitive(*op, cldnn::fully_connected(lstm_fc_id, fc_input_resh_id, WRreshapeID, bias.pid));
+
+            p.add_primitive(*op, cldnn::reshape(lstm_fc_resh_id, cldnn::input_info(lstm_fc_id), gemmSz));
+            p.add_primitive(*op, cldnn::reorder(lstm_fc_reor_id, cldnn::input_info(lstm_fc_resh_id), gemmLayout));
+            p.add_primitive(*op, cldnn::lstm_elt(lstm_elt_id, cldnn::input_info(lstm_fc_reor_id), cellStr, clip, 0, activations,
+                                                activation_params, cldnn::lstm_weights_order::fizo, 0));
+
+            hiddenStr = crop_id + ":hidden";
+            cellStr = crop_id + ":cell";
+            p.add_primitive(*op, cldnn::crop(hiddenStr, cldnn::input_info(lstm_elt_id), hiddenSz, cldnn::tensor{ 0, 0, 0, 0 }));
+            output_ids_offsets.push_back(cldnn::input_info(hiddenStr));
+
+            if (i < lstm_sequence_len - 1) {
+                p.add_primitive(*op, cldnn::crop(cellStr, cldnn::input_info(lstm_elt_id), hiddenSz, cellCropSz));
+            } else {
+                // last hidden state crop (output 2)
+
+                // last cell state crop (output 3)
+                p.add_primitive(*op, cldnn::crop(cellStr, cldnn::input_info(lstm_elt_id), hiddenSz, cellCropSz));
+            }
+        }
+
+        if (!isForward) std::reverse(output_ids_offsets.begin(), output_ids_offsets.end());
+        // concatenated hidden state (output 1)
+        cldnn::primitive_id concatStr = layerName + ":hiddenConcat";
+        p.add_primitive(*op, cldnn::concatenation(concatStr, output_ids_offsets, 1));
+
+        p.add_primitive(*op, cldnn::reshape(layerName + ".out0", concatStr, tensor_from_dims(op->get_output_shape(0))), {layerName});
+        p.add_primitive(*op, cldnn::reshape(layerName + ".out1", hiddenStr, tensor_from_dims(op->get_output_shape(1))));
+        p.add_primitive(*op, cldnn::reshape(layerName + ".out2", cellStr, tensor_from_dims(op->get_output_shape(2))));
+    } else {
+        if (op->get_input_shape(2).size() != 3 || op->get_input_shape(3).size() != 1 \
+            || op->get_input_shape(4).size() != 3 || op->get_input_shape(5).size() != 3 || op->get_input_shape(6).size() != 2)
+            OPENVINO_THROW("Wrong input shapes for LSTMSequence op ", op->get_friendly_name());
+        auto mutable_precision_firstsecond = op->get_output_element_type(1);
+        auto direction = op->get_direction();
+
+        if (p.use_new_shape_infer()) {
```

Comment:
I suggest enforcing new shape infer for model with LSTM cell/seq here https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/plugin/program_builder.cpp#L347 and just drop the code for legacy shape infer for this primitive

Reply:
done